File: packages/capacitor/src/eodin-event.ts
/**
 * Recommended event names for `EodinAnalytics.track()`.
 *
 * Aligns with `docs/logging/unified-event-reference.md` v1.1. Use this enum
 * for IDE autocomplete and cross-app consistency. Free-form string event
 * names continue to work for backward compatibility and for app-specific
 * domain events not covered here.
 *
 * @example
 * ```ts
 * import { EodinAnalytics, EodinEvent } from '@eodin/capacitor';
 *
 * await EodinAnalytics.track(EodinEvent.AppOpen);
 * await EodinAnalytics.track(EodinEvent.SubscribeStart, {
 *   plan: 'monthly',
 *   price: 9900,
 *   currency: 'KRW',
 * });
 *
 * // Free-form string still works:
 * await EodinAnalytics.track('recipe_view', { recipe_id: 'abc' });
 * ```
 */
export const EodinEvent = {
  // Lifecycle
  AppInstall: 'app_install',
  AppOpen: 'app_open',
  SessionResume: 'session_resume',
  SessionStart: 'session_start',
  SessionEnd: 'session_end',

  // Authentication / Identity
  SignUp: 'sign_up',
  SignIn: 'sign_in',
  SignOut: 'sign_out',
  AccountDelete: 'account_delete',

  // Onboarding
  OnboardingStart: 'onboarding_start',
  OnboardingStep: 'onboarding_step',
  OnboardingComplete: 'onboarding_complete',
  OnboardingSkip: 'onboarding_skip',

  // Core / Monetization
  CoreAction: 'core_action',
  PaywallView: 'paywall_view',
  PaywallDismiss: 'paywall_dismiss',
  SubscribeStart: 'subscribe_start',
  TrialStart: 'trial_start',
  SubscribeRenew: 'subscribe_renew',
  SubscriptionRestore: 'subscription_restore',
  IapPurchase: 'iap_purchase',
  DailyLimitReached: 'daily_limit_reached',
  DailyLimitDismiss: 'daily_limit_dismiss',
  DailyLimitUpgradeTap: 'daily_limit_upgrade_tap',
  PassView: 'pass_view',
  PassPurchase: 'pass_purchase',
  PassExpire: 'pass_expire',

  // Advertising — naming pattern: ad_<action> or ad_<format>_<action>
  AdRewardedView: 'ad_rewarded_view',
  AdInterstitialView: 'ad_interstitial_view',
  AdNativeView: 'ad_native_view',
  AdClick: 'ad_click',
  AdLoadFailed: 'ad_load_failed',
  AdFreePass: 'ad_free_pass',

  // Social / Viral
  Share: 'share',
  InviteShare: 'invite_share',
  InviteClaim: 'invite_claim',
  FriendAdd: 'friend_add',

  // iOS ATT
  AttPrompt: 'att_prompt',
  AttResponse: 'att_response',
} as const;

/** Union of recommended event names. Free-form string is also accepted. */
export type EodinEventName = (typeof EodinEvent)[keyof typeof EodinEvent];

File: packages/sdk-android/src/main/java/app/eodin/analytics/EodinEvent.kt
package app.eodin.analytics

/**
 * Recommended event names for [EodinAnalytics.track].
 *
 * Aligns with `docs/logging/unified-event-reference.md` v1.1. Use this enum
 * for IDE autocomplete and cross-app consistency. Free-form string event
 * names continue to work for backward compatibility and for app-specific
 * domain events not covered here.
 *
 * ### Example
 *
 * ```kotlin
 * EodinAnalytics.track(EodinEvent.APP_OPEN)
 * EodinAnalytics.track(
 *     EodinEvent.SUBSCRIBE_START,
 *     mapOf("plan" to "monthly", "price" to 9900, "currency" to "KRW")
 * )
 *
 * // Free-form string still works:
 * EodinAnalytics.track("recipe_view", mapOf("recipe_id" to "abc"))
 * ```
 */
enum class EodinEvent(val eventName: String) {
    // Lifecycle
    APP_INSTALL("app_install"),
    APP_OPEN("app_open"),
    SESSION_RESUME("session_resume"),
    SESSION_START("session_start"),
    SESSION_END("session_end"),

    // Authentication / Identity
    SIGN_UP("sign_up"),
    SIGN_IN("sign_in"),
    SIGN_OUT("sign_out"),
    ACCOUNT_DELETE("account_delete"),

    // Onboarding
    ONBOARDING_START("onboarding_start"),
    ONBOARDING_STEP("onboarding_step"),
    ONBOARDING_COMPLETE("onboarding_complete"),
    ONBOARDING_SKIP("onboarding_skip"),

    // Core / Monetization
    CORE_ACTION("core_action"),
    PAYWALL_VIEW("paywall_view"),
    PAYWALL_DISMISS("paywall_dismiss"),
    SUBSCRIBE_START("subscribe_start"),
    TRIAL_START("trial_start"),
    SUBSCRIBE_RENEW("subscribe_renew"),
    SUBSCRIPTION_RESTORE("subscription_restore"),
    IAP_PURCHASE("iap_purchase"),
    DAILY_LIMIT_REACHED("daily_limit_reached"),
    DAILY_LIMIT_DISMISS("daily_limit_dismiss"),
    DAILY_LIMIT_UPGRADE_TAP("daily_limit_upgrade_tap"),
    PASS_VIEW("pass_view"),
    PASS_PURCHASE("pass_purchase"),
    PASS_EXPIRE("pass_expire"),

    // Advertising — naming pattern: ad_<action> or ad_<format>_<action>
    AD_REWARDED_VIEW("ad_rewarded_view"),
    AD_INTERSTITIAL_VIEW("ad_interstitial_view"),
    AD_NATIVE_VIEW("ad_native_view"),
    AD_CLICK("ad_click"),
    AD_LOAD_FAILED("ad_load_failed"),
    AD_FREE_PASS("ad_free_pass"),

    // Social / Viral
    SHARE("share"),
    INVITE_SHARE("invite_share"),
    INVITE_CLAIM("invite_claim"),
    FRIEND_ADD("friend_add"),

    // iOS ATT (no-op on Android, kept for cross-platform consistency)
    ATT_PROMPT("att_prompt"),
    ATT_RESPONSE("att_response"),
}
